### PR TITLE
Atomics relaxation

### PR DESCRIPTION
**Pull request by rwaldron**

### CODE REVIEW

**Comment by anba** [comment]
I don't understand `testWithNonShareableBigIntTypedArrayConstructors` and `testWithNonShareableTypedArrayConstructors`:
- Both `BigInt64Array` and `BigUint64Array` can be used for all Atomics functions, except for `Atomics.wait`.
- Only `BigInt64Array` can be used for `Atomics.wait`; `BigUint64Array` will throw for `Atomics.wait`.
- `Int(8|16|32)Array` and `Uint(8|16|32)Array` can be used for all Atomics functions except for `wait`.
- Only `Int32Array` can be `wait`, `Int(8|16)Array` and `Uint(8|16|32)Array` throw for `wait`.

**Comment by rwaldron** [comment]
@anba corrections pushed.

File: test/built-ins/Atomics/xor/bigint/non-shared-bufferdata.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+esid: sec-atomics.xor
+description: >
+  Atomics.xor will operate on TA when TA.buffer is not a SharedArrayBuffer
+includes: [testBigIntTypedArray.js]
+features: [ArrayBuffer, Atomics, BigInt, TypedArray]
+---*/
+testWithBigIntTypedArrayConstructors(TA => {
+  const buffer = new ArrayBuffer(BigInt64Array.BYTES_PER_ELEMENT * 4);
+  const view = new TA(buffer);
+  assert.sameValue(Atomics.xor(i64a, 0, 1n), 0n, 'Atomics.xor(i64a, 0, 1n) returns 0n');
```
Comment (leobalter) [inline test/built-ins/Atomics/xor/bigint/non-shared-bufferdata.js:13]: missing `const i64a = new BigInt64Array(new ArrayBuffer(BigInt64Array.BYTES_PER_ELEMENT));`

File: test/built-ins/Atomics/add/bigint/non-shared-bufferdata.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+esid: sec-atomics.add
+description: >
+  Atomics.add will operate on TA when TA.buffer is not a SharedArrayBuffer
+includes: [testBigIntTypedArray.js]
+features: [ArrayBuffer, Atomics, BigInt, TypedArray]
+---*/
+testWithBigIntTypedArrayConstructors(TA => {
+  const buffer = new ArrayBuffer(BigInt64Array.BYTES_PER_ELEMENT * 4);
+  const view = new TA(buffer);
+  assert.sameValue(Atomics.add(i64a, 0, 1n), 0n, 'Atomics.add(i64a, 0, 1n) returns 0n');
```
Comment (leobalter) [inline test/built-ins/Atomics/add/bigint/non-shared-bufferdata.js:13]: missing `i64a` reference

File: test/built-ins/Atomics/and/bigint/non-shared-bufferdata.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+esid: sec-atomics.and
+description: >
+  Atomics.and will operate on TA when TA.buffer is not a SharedArrayBuffer
+includes: [testBigIntTypedArray.js]
+features: [ArrayBuffer, Atomics, BigInt, TypedArray]
+---*/
+testWithBigIntTypedArrayConstructors(TA => {
+  const buffer = new ArrayBuffer(BigInt64Array.BYTES_PER_ELEMENT * 4);
+  const view = new TA(buffer);
+  assert.sameValue(Atomics.and(i64a, 0, 1n), 0n, 'Atomics.and(i64a, 0, 1n) returns 0n');
```
Comment (leobalter) [inline test/built-ins/Atomics/and/bigint/non-shared-bufferdata.js:13]: missing reference for `i64a`, I believe you meant `view` instead?

File: test/built-ins/Atomics/compareExchange/bigint/non-shared-bufferdata.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+esid: sec-atomics.compareExchange
+description: >
+  Atomics.compareExchange will operate on TA when TA.buffer is not a SharedArrayBuffer
+includes: [testBigIntTypedArray.js]
+features: [ArrayBuffer, Atomics, BigInt, TypedArray]
+---*/
+testWithBigIntTypedArrayConstructors(TA => {
+  const buffer = new ArrayBuffer(BigInt64Array.BYTES_PER_ELEMENT * 4);
+  const view = new TA(buffer);
+  assert.sameValue(Atomics.compareExchange(i64a, 0, 0n, 1n), 0n, 'Atomics.compareExchange(i64a, 0, 0n, 1n) returns 0n');
```
Comment (leobalter) [inline test/built-ins/Atomics/compareExchange/bigint/non-shared-bufferdata.js:13]: same ref issue here

File: test/built-ins/Atomics/exchange/bigint/non-shared-bufferdata.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+esid: sec-atomics.exchange
+description: >
+  Atomics.exchange will operate on TA when TA.buffer is not a SharedArrayBuffer
+includes: [testBigIntTypedArray.js]
+features: [ArrayBuffer, Atomics, BigInt, TypedArray]
+---*/
+testWithBigIntTypedArrayConstructors(TA => {
+  const buffer = new ArrayBuffer(BigInt64Array.BYTES_PER_ELEMENT * 4);
+  const view = new TA(buffer);
+  assert.sameValue(Atomics.exchange(i64a, 0, 1n), 0n, 'Atomics.exchange(i64a, 0, 1n) returns 0n');
```
Comment (leobalter) [inline test/built-ins/Atomics/exchange/bigint/non-shared-bufferdata.js:13]: ref issue here

File: test/built-ins/Atomics/load/bigint/non-shared-bufferdata.js [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+esid: sec-atomics.load
+description: >
+  Atomics.load will operate on TA when TA.buffer is not a SharedArrayBuffer
+includes: [testBigIntTypedArray.js]
+features: [ArrayBuffer, Atomics, BigInt, TypedArray]
+---*/
+testWithBigIntTypedArrayConstructors(TA => {
+  const buffer = new ArrayBuffer(BigInt64Array.BYTES_PER_ELEMENT * 4);
+  const view = new TA(buffer);
+  assert.sameValue(Atomics.load(i64a, 0), 0n, 'Atomics.load(i64a, 0) returns 0n');
```
Comment (leobalter) [inline test/built-ins/Atomics/load/bigint/non-shared-bufferdata.js:13]: ref issue here

File: test/built-ins/Atomics/notify/bigint/non-shared-bufferdata-count-evaluation-throws.js [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+  valueOf() {
+    throw new Test262Error();
+  }
+};
+
+assert.throws(Test262Error, function() {
+  Atomics.notify(i64a, poisoned, 0);
+}, '`Atomics.notify(i64a, poisoned, 0)` throws Test262Error');
+
+
```
Comment (leobalter) [inline test/built-ins/Atomics/notify/bigint/non-shared-bufferdata-count-evaluation-throws.js:35]: nit: double empty lines

File: test/built-ins/Atomics/or/bigint/non-shared-bufferdata.js [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+esid: sec-atomics.or
+description: >
+  Atomics.or will operate on TA when TA.buffer is not a SharedArrayBuffer
+includes: [testBigIntTypedArray.js]
+features: [ArrayBuffer, Atomics, BigInt, TypedArray]
+---*/
+testWithBigIntTypedArrayConstructors(TA => {
+  const buffer = new ArrayBuffer(BigInt64Array.BYTES_PER_ELEMENT * 4);
+  const view = new TA(buffer);
+  assert.sameValue(Atomics.or(i64a, 0, 1n), 0n, 'Atomics.or(i64a, 0, 1n) returns 0n');
```
Comment (leobalter) [inline test/built-ins/Atomics/or/bigint/non-shared-bufferdata.js:13]: ref issue here

File: test/built-ins/Atomics/store/bigint/non-shared-bufferdata.js [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+esid: sec-atomics.store
+description: >
+  Atomics.store will operate on TA when TA.buffer is not a SharedArrayBuffer
+includes: [testBigIntTypedArray.js]
+features: [ArrayBuffer, Atomics, BigInt, TypedArray]
+---*/
+testWithBigIntTypedArrayConstructors(TA => {
+  const buffer = new ArrayBuffer(BigInt64Array.BYTES_PER_ELEMENT * 4);
+  const view = new TA(buffer);
+  assert.sameValue(Atomics.store(i64a, 0, 1n), 1n, 'Atomics.store(i64a, 0, 1n) returns 1n');
```
Comment (leobalter) [inline test/built-ins/Atomics/store/bigint/non-shared-bufferdata.js:13]: ref issue here

File: test/built-ins/Atomics/sub/bigint/non-shared-bufferdata.js [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+esid: sec-atomics.sub
+description: >
+  Atomics.sub will operate on TA when TA.buffer is not a SharedArrayBuffer
+includes: [testBigIntTypedArray.js]
+features: [ArrayBuffer, Atomics, BigInt, TypedArray]
+---*/
+testWithBigIntTypedArrayConstructors(TA => {
+  const buffer = new ArrayBuffer(BigInt64Array.BYTES_PER_ELEMENT * 4);
+  const view = new TA(buffer);
+  assert.sameValue(Atomics.store(i64a, 0, 1n), 1n, 'Atomics.store(i64a, 0, 1n) returns 1n');
```
Comment (leobalter) [inline test/built-ins/Atomics/sub/bigint/non-shared-bufferdata.js:13]: ref issue here

File: test/built-ins/Atomics/xor/bigint/non-shared-bufferdata.js [hunk 11]
```diff
@@ -0,0 +1,15 @@
...
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: sec-atomics.xor
+description: >
+  Atomics.xor will operate on TA when TA.buffer is not a SharedArrayBuffer
+includes: [testBigIntTypedArray.js]
+features: [ArrayBuffer, Atomics, BigInt, TypedArray]
+---*/
+testWithBigIntTypedArrayConstructors(TA => {
+  const buffer = new ArrayBuffer(BigInt64Array.BYTES_PER_ELEMENT * 4);
```
Comment (anba) [inline test/built-ins/Atomics/xor/bigint/non-shared-bufferdata.js:11]:
```suggestion
  const buffer = new ArrayBuffer(TA.BYTES_PER_ELEMENT * 4);
```

Also in all other places where this applies.

File: test/built-ins/Atomics/add/bigint/non-shared-bufferdata.js (hunk 2 above)
Comment (rwaldron) [reply test/built-ins/Atomics/add/bigint/non-shared-bufferdata.js:13]: Naw, it should be using `view`. I'm not sure how I messed that up on all of these test 🤦

Comment (leobalter) [reply test/built-ins/Atomics/add/bigint/non-shared-bufferdata.js]: I've problems like this where I left files unsaved in the code editor and then committed in progress changes. My guess is that you probably did something similar.

**Comment by rwaldron** [comment]
@leobalter thanks for thr ✅ , @anba thanks for the reviews

**Pull request merged**